docs/concepts/what-is-a-run-id.md

```diff
@@ -15,3 +15,9 @@ For example, a Temporal Cron Job is a chain of Workflow Executions that all have
 Each Workflow Execution within the chain is considered a _Run_.
 
 A Run Id uniquely identifies a Workflow Execution even if it shares a Workflow Id with other Workflow Executions.
+
+:::caution
+
+Run Id is unsafe to be used in a Workflow Execution for any logical choices because a reset operation can lead to non-determinism.
```

Comment:
Unsafe for which logical choices?

Pretty sure it is safe for logical choices when it comes to visibility and listing and signal from a client - etc... 

Also, I can not find the discussion that mentions this - wrong link maybe?